Sources/Sentry/SentryCrashIntegration.m

```diff
@@ -67,8 +72,17 @@ - (void)installWithOptions:(nonnull SentryOptions *)options
 {
     self.options = options;
 
+    SentryFileManager *fileManager = [[[SentrySDK currentHub] getClient] fileManager];
+    SentryAppStateManager *appStateManager = [[SentryAppStateManager alloc]
+            initWithOptions:options
+               crashAdapter:self.crashAdapter
+                fileManager:fileManager
+        currentDateProvider:[[SentryDefaultCurrentDateProvider alloc] init]
```

Comment:
A bit of a duplicate note but to be clear; do we need `alloc` here too?

Reply:
Maybe we should create a dependency container for common dependencies. We only need one crashAdapter, one CurrentDateProvider, ... . I updated the issue for that https://github.com/getsentry/sentry-cocoa/issues/1146